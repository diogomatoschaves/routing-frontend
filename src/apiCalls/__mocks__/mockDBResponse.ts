const mockDBRoute = { 
  "id": "1234qwer", 
  "eta": 123, 
  "provider": {
    "name" : "Das",
    "type" : "routing",
    "url": "routing.das.car",
  },
  "source": {
    "name" : "Das1",
  },

  "waypoints": {
    "origin": [ 1.5, 0.11 ],
    "destination": [ 34.5, -112.11 ],
    "middlePoints" : [ [-34.15, -112.12] ]
  },
  "geometry": {
    "type": "LineString", 
    "coordinates": [ 
      [ 34.5, 12.11 ],
      [ -34.5, -12.11 ],
      [ 34.5, 112.11 ],
      [ 34.5, -12.11 ] 
  ]},
  "ata": 345,
  "confidence": 3,
  "date": "How I can check this" 
}


export const mockValidDBResponse = {
  exists: true,
  routes: [mockDBRoute]
}

export const mockInvalidDBResponse = {
  exists: false,
  routes: []
}

