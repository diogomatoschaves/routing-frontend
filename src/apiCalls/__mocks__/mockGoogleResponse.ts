const mockGoogleResponse = {
  geocoded_waypoints: [],
  routes: [{
    legs: [{
      duration: 10,
      distance: 1000
    }],
    overview_polyline: "ujtbFbsagVLvOD|@PnDJx@JT^?"
  }],
  duration: 10,
  distance: 1000,
}

export default mockGoogleResponse
