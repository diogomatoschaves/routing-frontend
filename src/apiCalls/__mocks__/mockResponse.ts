export const mockCarResponse = {
  code: 'Ok',
  routes: [
    {
      totalDistance: 86.5,
      totalDuration: 15.1,
      legs: [
        {
          duration: 15.1,
          distance: 86.5,
          geometry: [
            { lon: 13.389869, lat: 52.510348 },
            { lon: 13.389899, lat: 52.51035 },
            { lon: 13.390113, lat: 52.510363 },
            { lon: 13.390178, lat: 52.510367 },
            { lon: 13.390305, lat: 52.510375 },
            { lon: 13.390589, lat: 52.510393 },
            { lon: 13.390867, lat: 52.510409 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 }
          ]
        }
      ]
    }
  ],
  locations: [
    {
      snapDistance: 0.45025736761464097,
      location: {
        lon: 13.389869,
        lat: 52.510348
      }
    },
    {
      snapDistance: 12.073291914193243,
      location: {
        lon: 13.39114,
        lat: 52.510425
      }
    }
  ]
}

export const mockCarTrafficResponse = {
  code: 'Ok',
  routes: [
    {
      totalDistance: 86.5,
      totalDuration: 12.6,
      legs: [
        {
          duration: 12.6,
          distance: 86.5,
          geometry: [
            { lon: 13.389869, lat: 52.510348 },
            { lon: 13.389899, lat: 52.51035 },
            { lon: 13.390113, lat: 52.510363 },
            { lon: 13.390178, lat: 52.510367 },
            { lon: 13.390305, lat: 52.510375 },
            { lon: 13.390589, lat: 52.510393 },
            { lon: 13.390867, lat: 52.510409 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 },
            { lon: 13.39114, lat: 52.510425 }
          ]
        }
      ]
    }
  ],
  locations: [
    {
      snapDistance: 0,
      location: {
        lon: 13.389869,
        lat: 52.510348
      }
    },
    {
      snapDistance: 0,
      location: {
        lon: 13.39114,
        lat: 52.510425
      }
    }
  ]
}

export const mockFootResponse = {
  code: 'Ok',
  routes: [
    {
      totalDistance: 709.8,
      totalDuration: 514.9,
      legs: [
        {
          duration: 514.9,
          distance: 709.8,
          geometry: [
            { lon: 13.387123, lat: 52.527155 },
            { lon: 13.387215, lat: 52.527166 },
            { lon: 13.387215, lat: 52.527166 },
            { lon: 13.387189, lat: 52.52703 },
            { lon: 13.387184, lat: 52.527008 },
            { lon: 13.387283, lat: 52.526386 },
            { lon: 13.387283, lat: 52.526386 },
            { lon: 13.387743, lat: 52.526289 },
            { lon: 13.387824, lat: 52.526272 },
            { lon: 13.388361, lat: 52.526146 },
            { lon: 13.388507, lat: 52.526111 },
            { lon: 13.388802, lat: 52.526039 },
            { lon: 13.388987, lat: 52.525933 },
            { lon: 13.389304, lat: 52.525859 },
            { lon: 13.389668, lat: 52.525767 },
            { lon: 13.389995, lat: 52.525692 },
            { lon: 13.389995, lat: 52.525692 },
            { lon: 13.390095, lat: 52.525772 },
            { lon: 13.390474, lat: 52.525863 },
            { lon: 13.393425, lat: 52.526381 },
            { lon: 13.393541, lat: 52.526401 },
            { lon: 13.393625, lat: 52.526417 },
            { lon: 13.393977, lat: 52.526485 },
            { lon: 13.395811, lat: 52.526839 },
            { lon: 13.395811, lat: 52.526839 },
            { lon: 13.395811, lat: 52.526839 }
          ]
        }
      ]
    }
  ],
  locations: [
    { snapDistance: 4.3196444235727816, location: { lon: 13.387123, lat: 52.527155 } },
    { snapDistance: 1.0574398354756074, location: { lon: 13.395811, lat: 52.526839 } }
  ]
}
