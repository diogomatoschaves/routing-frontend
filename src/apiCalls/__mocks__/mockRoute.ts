export const mockRoute = [
  { lat: 52.510348, lon: 13.389869 },
  { lat: 52.51035, lon: 13.389899 },
  { lat: 52.510363, lon: 13.390113 },
  { lat: 52.510367, lon: 13.390178 },
  { lat: 52.510375, lon: 13.390305 },
  { lat: 52.510393, lon: 13.390589 },
  { lat: 52.510409, lon: 13.390867 },
  { lat: 52.510425, lon: 13.39114 },
  { lat: 52.510425, lon: 13.39114 },
  { lat: 52.510425, lon: 13.39114 }
]

export const mockGoogleRoute = [
  { lat: 37.30107, lon: -121.91042 },
  { lat: 37.301, lon: -121.9131 },
  { lat: 37.30097, lon: -121.91341 },
  { lat: 37.30088, lon: -121.91429 },
  { lat: 37.30082, lon: -121.91458 },
  { lat: 37.30076, lon: -121.91469 },
  { lat: 37.3006, lon: -121.91469 }
]
