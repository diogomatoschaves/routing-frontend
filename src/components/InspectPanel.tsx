import React, { Fragment } from 'react'
import { Box, StyledDivider, StyledHeader } from '../styledComponents'
import {
  Body,
  GoogleResponse,
  HandleConfirmButton,
  HandleDeleteRoute,
  HandleValueUpdate,
  InputColors,
  InputValues,
  LocationInfo,
  MatchResponse,
  Option,
  OptionsHandler,
  OSRMRouteResponse,
  Route,
  UpdatePoint,
  UpdateState
} from '../types'
import { MAIN_PETROL } from '../utils/colours'
import EndpointRow from './EndpointRow'
import JsonBlock from './JsonBlock'
import ProfileToggler from './ProfileToggler'
import RoutesList from './RoutesList'
import Tabs from './Tabs'

interface Props {
  handleHideClick: (e: any) => void
  response: OSRMRouteResponse | GoogleResponse | MatchResponse
  responseOptionsHandler: OptionsHandler
  responseOption: Option
  body: Body
  endpointHandler: OptionsHandler
  updateState: UpdateState
  updatePoint: UpdatePoint
  handleValueUpdate: HandleValueUpdate
  locations: LocationInfo[]
  selectedService: number
  serviceOptions: any
  bodyEdit: boolean
  responseEdit: boolean
  debug: boolean
  inputValues: InputValues
  inputColors: InputColors
  addedRoutes: Route[]
  handleAddRoute: HandleConfirmButton
  handleClickRoute: HandleConfirmButton
  handleDeleteRoute: HandleDeleteRoute
  handleChangeBody: HandleConfirmButton
  handleCloseModal: HandleConfirmButton
  addDataTabsHandler: OptionsHandler
  modeTabsHandler: OptionsHandler
}

export default function InspectPanel(props: Props) {
  const {
    endpointHandler,
    updateState,
    handleValueUpdate,
    responseOption,
    response,
    body,
    selectedService,
    serviceOptions,
    responseEdit,
    bodyEdit,
    debug,
    addedRoutes,
    handleAddRoute,
    handleClickRoute,
    handleDeleteRoute,
    handleChangeBody,
    handleCloseModal,
    addDataTabsHandler,
    modeTabsHandler,
    responseOptionsHandler,
    inputValues,
    inputColors
  } = props

  return (
    <Box>
      <Box direction="row" justify="center" padding="0 0 10px 0">
        <Tabs
          tabsHandler={modeTabsHandler}
          updateState={updateState}
          id="modeTabsHandler"
          width="100px"
          justify="center"
        />
      </Box>
      <StyledDivider width="90%" />
      {debug ? (
        <RoutesList
          addedRoutes={addedRoutes}
          inputValues={inputValues}
          inputColors={inputColors}
          addDataTabsHandler={addDataTabsHandler}
          handleAddRoute={handleAddRoute}
          handleClickRoute={handleClickRoute}
          handleDeleteRoute={handleDeleteRoute}
          handleValueUpdate={handleValueUpdate}
          updateState={updateState}
        />
      ) : (
        <Fragment>
          <Box width="80%" padding="10px 0 10px 0">
            <StyledHeader overridecolor={MAIN_PETROL}>Request Endpoint</StyledHeader>
            <Box direction="row" justify="flex-start" padding="5px 0 10px 0">
              <EndpointRow updateState={updateState} endpointHandler={endpointHandler} />
            </Box>
          </Box>
          <Box width="80%" padding="10px 0 10px 0">
            <JsonBlock
              id={'body'}
              buttonId={'bodyEdit'}
              buttonText={'Confirm'}
              value={body}
              inputValues={inputValues}
              inputColors={inputColors}
              updateState={updateState}
              handleValueUpdate={handleValueUpdate}
              handleConfirmButton={handleChangeBody}
              className="body-textarea"
              text={'Request Body'}
              edit={bodyEdit}
              editable={true}
              title={'Edit Request Body'}
            />
          </Box>
          <Box width="80%" padding="10px 0 10px 0">
            <JsonBlock
              id={'response'}
              buttonId={'responseEdit'}
              buttonText={'Close'}
              value={response}
              inputValues={inputValues}
              inputColors={inputColors}
              updateState={updateState}
              handleConfirmButton={handleCloseModal}
              className="response-textarea"
              text={'Response'}
              edit={responseEdit}
              editable={false}
              title={'View Response'}
            />
          </Box>
        </Fragment>
      )}
    </Box>
  )
}
