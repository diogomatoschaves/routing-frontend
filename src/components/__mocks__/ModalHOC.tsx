import React from 'react'

export default function MockModalHOC (Component: any) {
  return (
    <div>
      <Component />
    </div>
  )
}