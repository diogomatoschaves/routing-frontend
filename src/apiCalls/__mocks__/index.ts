import { mockCarResponse, mockFootResponse, mockCarTrafficResponse } from './mockResponse'
import mockGoogleResponse from './mockGoogleResponse'
import { mockValidDBResponse, mockInvalidDBResponse } from './mockDBResponse'

export const routingApi = jest.fn().mockImplementation((profile, authorization, body) => {
  if (body.locations[0].lat === 100) {
    return Promise.resolve({
      ...mockCarResponse,
      code: 'LocationInvalid'
    })
  }

  return profile === 'car'
    ? Promise.resolve(mockCarResponse)
    : profile === 'car-traffic'
    ? Promise.resolve(mockCarTrafficResponse)
    : Promise.resolve(mockFootResponse)
})

export const googleDirections = jest
  .fn()
  .mockImplementation((google, profile, locations) => {
    if (locations[0].lat === 100) {
      return Promise.resolve({
        ...mockGoogleResponse,
        status: 'ZERO_RESULTS'
      })
    }
    return Promise.resolve(mockGoogleResponse)
  })

export const fetchRouteDB = jest.fn().mockImplementation(routeId => {
  return routeId === 'AAA'
    ? Promise.resolve(mockValidDBResponse)
    : Promise.resolve(mockInvalidDBResponse)
})
