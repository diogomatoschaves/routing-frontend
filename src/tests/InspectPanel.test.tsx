import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import TestRenderer from 'react-test-renderer'
import { mockCarResponse } from '../apiCalls/__mocks__/mockResponse'
import InspectPanel from '../components/InspectPanel'

const mockLocations = [
  {
    lat: null,
    lon: null,
    marker: 'map marker alternate',
    name: 'start',
    placeholder: 'Origin'
  },
  {
    lat: null,
    lon: null,
    marker: 'flag checkered',
    name: 'end',
    placeholder: 'Destination'
  }
]

const mockEventStart = { lon: 13.389869, lat: 52.510348 }
const mockEventEnd = { lon: 13.39114, lat: 52.510425 }

const mockBody = {
  locations: [
    {
      lat: mockEventStart.lat,
      lon: mockEventStart.lon
    },
    {
      lat: mockEventEnd.lat,
      lon: mockEventEnd.lon
    }
  ],
  reportGeometry: true
}

const serviceOptions = [
  {
    key: 'Route',
    text: 'Route',
    value: 0
  },
  {
    key: 'Match',
    text: 'Match',
    value: 1
  }
]

const mockEndpointHandler = {
  activeIdx: 0,
  options: [
    {
      key: 'develop',
      text: 'https://routing.develop.otonomousmobility.com/${PROFILE}',
      value: 0
    },
    {
      key: 'staging',
      text: 'https://routing.staging.otonomousmobility.com/${PROFILE}',
      value: 1
    },
    {
      key: 'testing',
      text: 'https://routing.testing.otonomousmobility.com/${PROFILE}',
      value: 2
    },
    { key: 'localhost', text: 'http://localhost:5000', value: 3 }
  ]
}

const mockAddDataTabsHandler = {
  activeIdx: 0,
  options: [
    { key: 'routingResponse', text: 'Routing Service', value: 0 },
    { key: 's3', text: 'Import from S3', value: 1 }
  ]
}

const mockModeTabsHandler = {
  activeIdx: 0,
  options: [
    { key: 'default', text: 'Interactive', value: 0 },
    { key: 'debug', text: 'Debugging', value: 1 }
  ]
}

const mockResponseOptionsHandler = {
  activeIdx: 0,
  options: [
    { key: 'routeResponse', text: 'No Traffic', value: 0 },
    { key: 'trafficResponse', text: 'w/ Traffic', value: 1 },
    { key: 'googleResponse', text: 'Google', value: 2 }
  ]
}

const inputValues = {
  body: '',
  match: '',
  response: '',
  route: ''
}

const inputColors = {
  body: 'rgb(100, 100, 100)',
  match: 'rgb(100, 100, 100)',
  route: 'rgb(100, 100, 100)'
}

it('expect to render InspectPanel component', () => {
  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={['/']}>
      <InspectPanel
        inputValues={inputValues}
        inputColors={inputColors}
        responseEdit={false}
        bodyEdit={false}
        debug={false}
        addedRoutes={[]}
        addDataTabsHandler={mockAddDataTabsHandler}
        modeTabsHandler={mockModeTabsHandler}
        handleHideClick={jest.fn()}
        handleAddRoute={jest.fn()}
        handleChangeBody={jest.fn()}
        handleCloseModal={jest.fn()}
        handleValueUpdate={jest.fn()}
        handleDeleteRoute={jest.fn()}
        handleClickRoute={jest.fn()}
        response={mockCarResponse}
        body={mockBody}
        endpointHandler={mockEndpointHandler}
        updatePoint={jest.fn()}
        updateState={jest.fn()}
        responseOptionsHandler={mockResponseOptionsHandler}
        responseOption={
          mockResponseOptionsHandler.options[mockResponseOptionsHandler.activeIdx]
        }
        locations={mockLocations}
        selectedService={0}
        serviceOptions={serviceOptions}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
})
