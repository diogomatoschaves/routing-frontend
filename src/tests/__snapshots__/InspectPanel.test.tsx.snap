// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render InspectPanel component 1`] = `
<div
  className="sc-bdVaJa eBppGN"
>
  <div
    className="sc-bdVaJa dcMlZW"
    direction="row"
  >
    <div
      className="sc-bdVaJa kygzpb"
      direction="row"
    >
      <div
        className="sc-bdVaJa sc-eNQAEJ ibiofc"
        color={true}
        id="default"
        onClick={[Function]}
        width="100px"
      >
        Interactive
      </div>
      <div
        className="sc-bdVaJa sc-eNQAEJ fTBrZq"
        color={false}
        id="debug"
        onClick={[Function]}
        width="100px"
      >
        Debugging
      </div>
    </div>
  </div>
  <div
    className="ui divider sc-gzVnrw fYaiCA"
    width="90%"
  />
  <div
    className="sc-bdVaJa dNMDiv"
    direction="row"
    height="70px"
  >
    <div
      className="sc-bdVaJa gOaCBL"
      width="40%"
    >
      <div
        className="sc-bdVaJa gxFsig"
        direction="row"
        height="45px"
      >
        <div
          className="sc-bdVaJa clGjkA"
          onClick={[Function]}
          width="10%"
        >
          <i
            aria-hidden="true"
            className="grey angle left large icon sc-kgoBCf fROEoW"
            onClick={[Function]}
          />
        </div>
        <div
          className="sc-bdVaJa gNhOwl"
          width="60%"
        >
          <div
            className="ui large label sc-kGXeez bnCwyW"
            onClick={[Function]}
          >
            No Traffic
          </div>
        </div>
        <div
          className="sc-bdVaJa clGjkA"
          onClick={[Function]}
          width="10%"
        >
          <i
            aria-hidden="true"
            className="grey angle right large icon sc-kgoBCf fROEoW"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa fEcaVD"
    width="80%"
  >
    <div
      className="ui header sc-htoDjs eoOxOA"
      overridecolor="#00677f"
    >
      Request Endpoint
    </div>
    <div
      className="sc-bdVaJa fuiAsE"
      direction="row"
    >
      <div
        aria-expanded={false}
        className="ui fluid search selection dropdown sc-EHOje bmMNag"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <input
          aria-autocomplete="list"
          autoComplete="off"
          className="search"
          onChange={[Function]}
          style={
            Object {
              "width": undefined,
            }
          }
          tabIndex={0}
          type="text"
          value=""
        />
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          https://routing.develop.otonomousmobility.com/\${PROFILE}
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
          role="listbox"
        >
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              https://routing.develop.otonomousmobility.com/\${PROFILE}
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              https://routing.staging.otonomousmobility.com/\${PROFILE}
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              https://routing.testing.otonomousmobility.com/\${PROFILE}
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              http://localhost:5000
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa fEcaVD"
    width="80%"
  >
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
    >
      <div
        className="ui header sc-htoDjs eoOxOA"
        overridecolor="#00677f"
      >
        Request Body
      </div>
      <button
        className="ui button body-textarea sc-chPdSV bfyvDP"
        id="bodyEdit"
        marginleft="10px"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
      width="100%"
    >
      <div
        className="sc-jzJRlG kJvmHz"
      >
        <div
          className="sc-fjdhpX cwkpjk"
          onClick={[Function]}
        >
          
          <i
            aria-hidden="true"
            className="caret down icon"
            onClick={[Function]}
          />
        </div>
        <div
          className="sc-jzJRlG kJvmHz"
        >
          <div
            className="sc-fjdhpX cwkpjk"
            onClick={[Function]}
          >
            locations
            <i
              aria-hidden="true"
              className="caret down icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="sc-jzJRlG kJvmHz"
          >
            <div
              className="sc-fjdhpX cwkpjk"
              onClick={[Function]}
            >
              0
              <i
                aria-hidden="true"
                className="caret down icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                lat
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                52.510348
              </span>
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                lon
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                13.389869
              </span>
            </div>
          </div>
          <div
            className="sc-jzJRlG kJvmHz"
          >
            <div
              className="sc-fjdhpX cwkpjk"
              onClick={[Function]}
            >
              1
              <i
                aria-hidden="true"
                className="caret down icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                lat
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                52.510425
              </span>
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                lon
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                13.39114
              </span>
            </div>
          </div>
        </div>
        <div
          className="sc-jzJRlG kJvmHz"
        >
          <span
            className="sc-cSHVUG gvhixb"
          >
            reportGeometry
            : 
          </span>
          <span
            className="sc-kAzzGY jdxiPe"
          >
            true
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-VigVT fTAign header"
    >
      Edit Request Body
    </div>
    <div
      className="sc-jTzLTM chFRNh content"
    >
      <form
        className="ui form sc-gZMcBi hTYlPb"
        color="rgb(100, 100, 100)"
        focuscolor="#5097ab"
        onSubmit={[Function]}
      >
        <textarea
          className="textarea"
          id="body"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          placeholder=""
          rows={15}
          value=""
        />
      </form>
    </div>
    <div
      className="actions"
    >
      <div
        className="sc-bdVaJa fXWXpp"
        direction="row"
      >
        <button
          alignend={true}
          backgroundcolor="#00677f"
          className="ui button sc-bZQynM iSeNLM"
          onClick={[Function]}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa fEcaVD"
    width="80%"
  >
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
    >
      <div
        className="ui header sc-htoDjs eoOxOA"
        overridecolor="#00677f"
      >
        Response
      </div>
      <button
        className="ui button response-textarea sc-chPdSV bfyvDP"
        id="responseEdit"
        marginleft="10px"
        onClick={[Function]}
      >
        View source
      </button>
    </div>
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
      width="100%"
    >
      <div
        className="sc-jzJRlG kJvmHz"
      >
        <div
          className="sc-fjdhpX cwkpjk"
          onClick={[Function]}
        >
          
          <i
            aria-hidden="true"
            className="caret down icon"
            onClick={[Function]}
          />
        </div>
        <div
          className="sc-jzJRlG kJvmHz"
        >
          <span
            className="sc-cSHVUG gvhixb"
          >
            code
            : 
          </span>
          <span
            className="sc-kAzzGY jdxiPe"
          >
            Ok
          </span>
        </div>
        <div
          className="sc-jzJRlG kJvmHz"
        >
          <div
            className="sc-fjdhpX cwkpjk"
            onClick={[Function]}
          >
            routes
            <i
              aria-hidden="true"
              className="caret down icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="sc-jzJRlG kJvmHz"
          >
            <div
              className="sc-fjdhpX cwkpjk"
              onClick={[Function]}
            >
              0
              <i
                aria-hidden="true"
                className="caret down icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                totalDistance
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                86.5
              </span>
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                totalDuration
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                15.1
              </span>
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <div
                className="sc-fjdhpX cwkpjk"
                onClick={[Function]}
              >
                legs
                <i
                  aria-hidden="true"
                  className="caret down icon"
                  onClick={[Function]}
                />
              </div>
              <div
                className="sc-jzJRlG kJvmHz"
              >
                <div
                  className="sc-fjdhpX cwkpjk"
                  onClick={[Function]}
                >
                  0
                  <i
                    aria-hidden="true"
                    className="caret down icon"
                    onClick={[Function]}
                  />
                </div>
                <div
                  className="sc-jzJRlG kJvmHz"
                >
                  <span
                    className="sc-cSHVUG gvhixb"
                  >
                    duration
                    : 
                  </span>
                  <span
                    className="sc-kAzzGY jdxiPe"
                  >
                    15.1
                  </span>
                </div>
                <div
                  className="sc-jzJRlG kJvmHz"
                >
                  <span
                    className="sc-cSHVUG gvhixb"
                  >
                    distance
                    : 
                  </span>
                  <span
                    className="sc-kAzzGY jdxiPe"
                  >
                    86.5
                  </span>
                </div>
                <div
                  className="sc-jzJRlG kJvmHz"
                >
                  <div
                    className="sc-fjdhpX cwkpjk"
                    onClick={[Function]}
                  >
                    geometry
                    <i
                      aria-hidden="true"
                      className="caret right icon"
                      onClick={[Function]}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-jzJRlG kJvmHz"
        >
          <div
            className="sc-fjdhpX cwkpjk"
            onClick={[Function]}
          >
            locations
            <i
              aria-hidden="true"
              className="caret down icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="sc-jzJRlG kJvmHz"
          >
            <div
              className="sc-fjdhpX cwkpjk"
              onClick={[Function]}
            >
              0
              <i
                aria-hidden="true"
                className="caret down icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                snapDistance
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                0.45025736761464097
              </span>
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <div
                className="sc-fjdhpX cwkpjk"
                onClick={[Function]}
              >
                location
                <i
                  aria-hidden="true"
                  className="caret right icon"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="sc-jzJRlG kJvmHz"
          >
            <div
              className="sc-fjdhpX cwkpjk"
              onClick={[Function]}
            >
              1
              <i
                aria-hidden="true"
                className="caret down icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <span
                className="sc-cSHVUG gvhixb"
              >
                snapDistance
                : 
              </span>
              <span
                className="sc-kAzzGY jdxiPe"
              >
                12.073291914193243
              </span>
            </div>
            <div
              className="sc-jzJRlG kJvmHz"
            >
              <div
                className="sc-fjdhpX cwkpjk"
                onClick={[Function]}
              >
                location
                <i
                  aria-hidden="true"
                  className="caret right icon"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-VigVT fTAign header"
    >
      View Response
    </div>
    <div
      className="sc-jTzLTM chFRNh content"
    >
      <form
        className="ui form sc-gZMcBi hTYlPb"
        color="rgb(100, 100, 100)"
        focuscolor="#5097ab"
        onSubmit={[Function]}
      >
        <textarea
          className="textarea"
          id="response"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          placeholder=""
          rows={15}
          value=""
        />
      </form>
    </div>
    <div
      className="actions"
    >
      <div
        className="sc-bdVaJa fXWXpp"
        direction="row"
      >
        <button
          alignend={true}
          backgroundcolor="#00677f"
          className="ui button sc-bZQynM iSeNLM"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
